src/sp/it/pl/util/file/Util.kt

```diff
@@ -72,24 +72,36 @@ fun File.isAnyChildOf(parent: File) = parent.isAnyParentOf(this)
  * @return child files of the directory or empty if parameter null, not a directory or I/O error occurs
  */
 @Suppress("DEPRECATION")
-fun File.listChildren(): Stream<File> = listFiles()?.asSequence()?.asStream() ?: Stream.empty()
+fun File.listChildren(): Stream<File> = 
+        listFiles().asStream()
 
 @Suppress("DEPRECATION")
-fun File.seqChildren(): Sequence<File> = listFiles()?.asSequence() ?: sequenceOf()
+fun File.listChildren(filter: FileFilter): Stream<File> = 
+        listFiles(filter).asStream()
 
 @Suppress("DEPRECATION")
-fun File.listChildren(filter: FileFilter): Stream<File> = listFiles(filter)?.asSequence()?.asStream() ?: Stream.empty()
+fun File.listChildren(filter: FilenameFilter): Stream<File> =
+        listFiles(filter).asStream()
 
 @Suppress("DEPRECATION")
-fun File.listChildren(filter: FilenameFilter): Stream<File> = listFiles(filter)?.asSequence()?.asStream() ?: Stream.empty()
+fun File.seqChildren(): Sequence<File> =
+        listFiles()?.asSequence().orEmpty()
+
+/** @return a Stream built from this array, which is empty if the array is null */
+fun <T> Array<T>?.asStream() = this?.asSequence()?.asStream() ?: Stream.empty()
```

Comment:
This is a horrible general purpose method.

The naming does not denote there is any null handling at all and the nullable receiver type means this information is completely lost to the reader at the call site.

AsStreamOrEmpty() alternative that begs to be used is verbose and largely unnecessary. Passing around null arrays or collection is clear antipattern we do not want to instigate in any way (like by making it more comfortable for the dev). 

We also do not want to prod developers to use Arrays and Streams so these would be a bridging methods asking to be removed in the future.

In general orEmpty() methods are questionable as they introduce implicit behavior (very similar to that of the operator overloading), but such method with an added conversion on top introduce another complexity in the implicit behavior which only broadens the cognitive depth for the reader.

Even if, methods with nullable receiver themselves are already questionable enough, as they take away expressive power of the Elvis operator they end up substituting, i.e., `nullableObject.invoke()` will always be inferior to explicit `nullableObject?.invoke() ?: fallbackValue`. The general rule of thumb (I go by) is that methods are defined with non null receiver (just like instance methods on an object) and the null passing/handling is to be left to the developer. Not doing brings about hidden null handling, unexpected behavior in form of default values and failure of error handling from not being aware of there being an error case in the first place.

Convenience methods always have to prove themselves against the added complexity they introduce.

Lastly, I stand by my opinion that coincidental code duplicity is not to be refactored into method, like in this very case.

This is a clear case of less is more. Please remove the methods completely or at the very least make them private. I support the former because even as private, the methods already bring more problems than they solve (confusion, potential to abuse/misuse, javadoc, potential bugs, etc) and the refactored code can still remain as a one liner. We should strive to reduce the number of util/toplevel methods by relying on offocial API and clever strategically written ones and avoid methods to reduce 1 word somewhere in the project.

Reply:
True, they just hit me as something that's not obeying DRY. I'll revert it.